src/core/src/main/java/org/apache/jmeter/testelement/TestPlan.java

```diff
@@ -82,12 +85,23 @@ public void prepareForPreCompile()
     }
 
     /**
-     * Fetches the functional mode property
-     *
+     * Fetches the functional mode property<br>
+     * Could be change for no-GUI test with jmeter property: {@code PROP_FUNCTIONAL_MODE}
      * @return functional mode
      */
     public boolean isFunctionalMode() {
-        return getPropertyAsBoolean(FUNCTIONAL_MODE);
+        boolean functionalModeDefault = getPropertyAsBoolean(FUNCTIONAL_MODE);
+        log.debug("functionalModeDefault=" + functionalModeDefault);
+        boolean functionalModeReturn = functionalModeDefault;
+        if (isNonGui()) {
```

Comment:
Why the override is for `non-gui-mode` only?
I think we should not add mode-dependent settings as it would be hard to tell why the behaviour is different.

Reply:
Why the override is for non-gui-mode only?
Because Gui mode you could see is the check box is check or not and check if you want to modify : click in check box.
With non-gui mode you force the value without change the jmeter script. 

In gui mode do you display the check box as it's set in the script script or as is overriding with external property ?
And what is current value ? it is you set external property and click in the check box ? 
Who overriding the check box value ? The external property or the last click with the gui on the check box ?
I don't have the solution for gui mode.

Reply:
That is fair indeed. It looks like we can support GUI if we implement a UI for "checkbox that can be configured with an expression"